src/fsharp/FSharp.Core/array.fs

```diff
@@ -559,6 +559,31 @@ namespace Microsoft.FSharp.Collections
                     res.[i] <- Microsoft.FSharp.Primitives.Basics.Array.subUnchecked i windowSize array
                 res
 
+        [<CompiledName("ChunksOf")>]
+        let chunksOf chunkSize (array:'T[]) =
+            checkNonNull "array" array
+            if chunkSize <= 0 then invalidArg "chunkSize" (SR.GetString(SR.inputMustBeNonNegative))
```

Comment:
:+1: I knew it wasn't quite right, but it was used in windowed with the same check and I assumed there was more to it than just adding a new constant :)